### PR TITLE
Add Gateway and REST model documentation and updates.

### PR DESCRIPTION
**Pull request by AntiTcb**

### CODE REVIEW

File: src/Wumpus.Net.Core/Entities/Permissions/ChannelPermissions.cs [hunk 1]
```diff
@@ -2,37 +2,58 @@
...
-        //ChangeNickname = 1 << 26,
-        //ManageNicknames = 1 << 27,
-        ManagePermissions = 1 << 28,
-        ManageWebhooks = 1 << 29,
-        //ManageEmojis = 1 << 30
+        None                = 0,
+
+        // General
+        /// <summary> Allow creation of instant <see cref="Invite"/>s. </summary>
+        CreateInstantInvite = 0x00_00_00_01,
```
Comment (RogueException) [inline src/Wumpus.Net.Core/Entities/Permissions/ChannelPermissions.cs]: This format is inconsistent with some of the other flags enums. I don't see any benefit to this over the old bit positions but everything should be consistent.

File: src/Wumpus.Net.Gateway/Events/GuildSyncEvent.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 namespace Wumpus.Events
 {
+    // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/Events/GuildSyncEvent.cs]: Yeah, lets take this out.

File: src/Wumpus.Net.Gateway/Events/RecipientEvent.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 namespace Wumpus.Events
 {
+    // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/Events/RecipientEvent.cs]: Any undocumented stuff lets just rip out for now

File: src/Wumpus.Net.Gateway/GatewayOpCode.cs [hunk 4]
```diff
@@ -27,6 +31,7 @@ public enum GatewayOpCode : byte
         Hello = 10,
         /// <summary> C←S - Used to reply to a client's heartbeat. </summary>
         HeartbeatAck = 11,
+        // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/GatewayOpCode.cs]: RIP

File: src/Wumpus.Net.Rest/DiscordRestConstants.cs [hunk 5]
```diff
@@ -16,6 +16,7 @@ public static class DiscordRestConstants
         public const string CDNUrl = "https://cdn.discordapp.com/";
         public const string InviteUrl = "https://discord.gg/";
         
+        // TODO: Move these to models, or in Preconditions?
```
Comment (RogueException) [inline src/Wumpus.Net.Rest/DiscordRestConstants.cs]: Seems all constraints are being moved to models?

File: src/Wumpus.Net.Core/Entities/Color.cs [hunk 6]
```diff
@@ -92,6 +94,13 @@ public Color(float r, float g, float b)
                 (uint)(b * 255.0f);
         }
 
+#if NETSTANDARD2_0 || NET45
```
Comment (RogueException) [inline src/Wumpus.Net.Core/Entities/Color.cs]: Good for this PR but we should API review this later

File: src/Wumpus.Net.Gateway/GatewayDispatchType.cs [hunk 7]
```diff
@@ -2,41 +2,75 @@
...
         [ModelEnumValue("READY")] Ready,
+        /// <summary> Lazy-load for unavailable <see cref="Entities.Guild"/>, <see cref="Entities.Guild"/> became available, or a <see cref="Entities.User"/> joined a new <see cref="Entities.Guild"/>. </summary>
         [ModelEnumValue("GUILD_CREATE")] GuildCreate,
+        /// <summary> <see cref="Entities.Guild"/> was updated. </summary>
         [ModelEnumValue("GUILD_UPDATE")] GuildUpdate,
+        /// <summary> <see cref="Entities.Guild"/> became unavailable, or <see cref="Entities.User"/> left/was removed from a <see cref="Entities.Guild"/>. </summary>
         [ModelEnumValue("GUILD_DELETE")] GuildDelete,
+        /// <summary> <see cref="Entities.Guild"/> <see cref="Entities.Emoji"/>s were updated. </summary>
         [ModelEnumValue("GUILD_EMOJIS_UPDATE")] GuildEmojisUpdate,
+        // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/GatewayDispatchType.cs]: RIP

File: src/Wumpus.Net.Gateway/GatewayDispatchType.cs [hunk 8]
```diff
@@ -2,41 +2,75 @@
...
         [ModelEnumValue("CHANNEL_DELETE")] ChannelDelete,
+        /// <summary> New <see cref="Entities.User"/> joined <see cref="Entities.Guild"/>. </summary>
         [ModelEnumValue("GUILD_MEMBER_ADD")] GuildMemberAdd,
+        /// <summary> <see cref="Entities.GuildMember"/> was updated. </summary>
         [ModelEnumValue("GUILD_MEMBER_UPDATE")] GuildMemberUpdate,
+        /// <summary> <see cref="Entities.User"/> was removed from a <see cref="Entities.Guild"/>. </summary>
         [ModelEnumValue("GUILD_MEMBER_REMOVE")] GuildMemberRemove,
+        /// <summary> Response to <see cref="GatewayOpCode.RequestGuildMembers"/>. </summary>
         [ModelEnumValue("GUILD_MEMBERS_CHUNK")] GuildMembersChunk,
+        // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/GatewayDispatchType.cs]: RIP

File: src/Wumpus.Net.Gateway/GatewayDispatchType.cs [hunk 9]
```diff
@@ -2,41 +2,75 @@
...
         [ModelEnumValue("CHANNEL_RECIPIENT_ADD")] ChannelRecipientAdd,
+        // TODO: Remove, undocumented?
```
Comment (RogueException) [inline src/Wumpus.Net.Gateway/GatewayDispatchType.cs]: RIP

File: src/Wumpus.Net.Core/Entities/Permissions/ChannelPermissions.cs (hunk 1 above)
Comment (AntiTcb) [reply src/Wumpus.Net.Core/Entities/Permissions/ChannelPermissions.cs]: The only benefit I considered was being able to eyeball the hexadecimal value at a glance, as Discord's docs list out the permission values in that format. If consistency is still desired in this case, I'll swap it back to the bit shifts.

Comment (RogueException) [reply src/Wumpus.Net.Core/Entities/Permissions/ChannelPermissions.cs]: Ah, if it's that way in the docs, we could use the same format. In that case I'd prob get rid of the separators.

**Comment by RogueException** [comment]
Merged in https://github.com/RogueException/Wumpus.Net/commit/b54b3268eac85d7cf9f33157993ec768d6f17e17

**Pull request closed without merging**